pkg/reconciler/managed/reconciler.go

```diff
@@ -306,6 +306,12 @@ type ExternalObservation struct {
 	// unless an existing key is overwritten. Crossplane may publish these
 	// credentials to a store (e.g. a Secret).
 	ConnectionDetails ConnectionDetails
+
+	// ObservationMessage is a Debug level message that is sent to the
+	// reconciler when there is a change in the observed Managed Resource.
+	// It is useful for finding what change there is between what was
+	// observed and the resource's definition.
+	ObservationMessage string
```

Comment:
What do you think about naming this something like `Diff` and providing guidance that it should be a `cmp.Diff` between the observed and desired state? The idea being that we'd be a little more opinionated about how this field was used.

Reply:
I like this idea. The cmp.Diff is definitely the right direction.